canine/backends/base.py

```diff
@@ -404,11 +412,15 @@ def wait_for_cluster_ready(self):
         Blocks until the main partition is marked as up
         """
         df = self.sinfo()
-        default = [value for value in df.index if value.endswith('*')]
+        default = df.index[df.index.str.contains(r"\*$")]
+
+        # wait for partition to appear
         while len(default) == 0:
             time.sleep(10)
             df = self.sinfo()
-            default = [value for value in df.index if value.endswith('*')]
-        while (df.AVAIL[default] != 'up').all():
+            default = df.index[df.index.str.contains(r"\*$")]
+
+        # wait for any node in the partition to be ready
+        while ~df.loc[default, "STATE"].str.contains(r"(?:mixed|idle|completing|allocated\+?)$").any():
```

Comment:
This doesn't actually work for the TransientGCP backend. That backend marks the whole partition as `AVAIL = down` until it's ready. However the nodes themselves are listed as `idle~`. In my tests, canine just sits there forever

Reply:
OK, I'll modify the regex to exclude any suffixes from here: https://slurm.schedmd.com/sinfo.html#SECTION_NODE-STATE-CODES

A side effect of this would be that user-managed clusters with all nodes in power save mode would never be marked as "up" until the user manually starts some of them.

Reply:
On second thought, trying to cover all different use cases here is confusing. 

I think it would be better to have the TransientGCP backend override this class (with how the base implementation was previously), since AFAIK it's the only one that explicitly depends on the partition status as a signal for availability.

Reply:
Resolved in acca6da.